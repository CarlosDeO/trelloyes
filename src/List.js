import React from 'react';
import './List.css';
import Card from './Card';

class List extends React.Component {
    render() {
        return (
            <section className="List">
            <header className="List-header"></header>
            <div className='List-cards'>
        {this.props.cards.map((card) =>
          <Card
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            onClickDelete={this.props.onClickDelete}
          />
        )}
            <button 
            type='button'
            className='List-add-button'
            onClick={() => this.props.onClickAdd(this.props.id)}
             >
          + Add Random Card
        </button>
            </div>
            </section>
        )
    }
}

List.defaultProps = {
    onClickAdd: () => {},
  }

export default List;