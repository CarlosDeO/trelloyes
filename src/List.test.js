import React from 'react';
import ReactDOM from 'react-dom';
import List from './Card';
import renderer from 'react-test-renderer';


describe('List component', () => {

    const lists = [
        {
          id: '1',
          header: 'First list',
          cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ],
        },
        {
          id: '2',
          header: 'Second list',
          cardIds: ['b', 'c', 'd', 'f', 'h', 'i', 'k'],
        },
        {
          id: '3',
          header: 'Third list',
          cardIds: [ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ],
        },
        {
          id: '4',
          header: 'Fourth list',
          cardIds: [ 'l', 'm' ],
        },
    ]

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<List />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
        const tree = renderer
        .create(<List
            key={lists[0].id}
            header={lists[0].header}
            // cards={lists[0].cardIds.map(id => store.allCards[id])}
          />)
          .toJSON();
          expect(tree).toMatchSnapshot();
    });
});