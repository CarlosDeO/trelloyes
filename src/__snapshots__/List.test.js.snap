// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component render the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
  <h3>
     
     
  </h3>
  <p>
     
     
  </p>
</div>
`;
